runtime/src/main/java/org/corfudb/runtime/CorfuRuntime.java

```diff
@@ -845,14 +846,17 @@ public IClientRouter getRouter(String address) {
      * If the layout has been previously invalidated and a new layout has not yet been retrieved,
      * this function does nothing.
      */
-    public synchronized void invalidateLayout() {
+    public synchronized CompletableFuture<Layout> invalidateLayout() {
```

Comment:
changing the the public api. have you checked the implications?

Reply:
yep, it doesn't affect anything